### PR TITLE
Routing cleanup

### PR DESCRIPTION
**Pull request by michaelrkn**
* Remove large tables in favor of shorter explanations
* Provide more examples
* Remove redundant information
* Remove many (but not all) references to controllers
* Remove the page on "The Application Template" since it is more about routing

This is just a start - there is plenty more work to be done on this section, especially on removing stuff about controllers.

### CODE REVIEW

File: source/configuring-ember/debugging.md [hunk 1]
```diff
@@ -34,13 +29,6 @@ ENV.APP.LOG_VIEW_LOOKUPS = true;
 Ember.keys(Ember.TEMPLATES)
 ```
 
-#### Handlebars Debugging Helpers
```
Comment (locks) [inline source/configuring-ember/debugging.md]: aren't these removed in another PR?

**Comment by locks** [comment]
This PR seems to have changes other than the routing, is that assessment correct or am I just getting lost in the changes?

**Comment by michaelrkn** [comment]
Sorry, git weirdness. Fixed now - can you take another look?

**Comment by locks** [comment]
I don't know, people like the tables as it's very straightforward to understand the routes generated. Maybe make a reference to the Inspector and link there?

**Comment by michaelrkn** [comment]
I am biased anti-table, especially because this page was just overloaded with them. Maybe could you leave an inline comment any place that you think a table would be easier to understand than the text?

File: source/routing/defining-your-routes.md [hunk 2]
```diff
@@ -45,335 +45,48 @@ the case of `/`, the name `index`).
...
-
-<small><sup>2</sup> `:post_id` is the post's id.  For a post with id = 1, the route will be:
-`/post/1`</small>
-
-The `comments` template will be rendered in the `post` outlet.
-All templates under `comments` (`comments/index` and `comments/new`) will be rendered in the `comments` outlet.
-
-The route, controller, and view class names for the comments resource are not prefixed with `Post`.
-
-### Multi-word Model Names
```
Comment (locks) [inline source/routing/defining-your-routes.md]: is this really not necessary?

File: source/routing/defining-your-routes.md [hunk 3]
```diff
@@ -45,335 +45,48 @@ the case of `/`, the name `index`).
...
 
 A nested route's names includes the names of its ancestors.
 If you want to transition to a route (either
 via `transitionTo` or `{{#link-to}}`), make sure to use the full route
 name (`posts.new`, not `new`).
 
-That said, transitioning to `posts` or creating a link to `posts` is equivalent
-to transitioning to `posts.index` or linking to `posts.index`.
-
-Visiting `/` renders the `index` template, as you would expect.
```
Comment (locks) [inline source/routing/defining-your-routes.md]: is this information somewhere else? I think it's an important tidbit

File: source/routing/defining-your-routes.md (hunk 2 above)
Comment (michaelrkn) [inline source/routing/defining-your-routes.md]: Why is it necessary? What does following that convention get you?

File: source/routing/defining-your-routes.md (hunk 3 above)
Comment (michaelrkn) [inline source/routing/defining-your-routes.md]: Yeah, https://github.com/emberjs/guides/pull/566/files#diff-93ea9413dddafcd861a10410813e32d4R138

File: source/routing/defining-your-routes.md (hunk 2 above)
Comment (locks) [inline source/routing/defining-your-routes.md]: generated Route model hook

Comment (michaelrkn) [inline source/routing/defining-your-routes.md]: is this documented somewhere? i vaguely remember reading about it, but i can't find it.

File: source/routing/defining-your-routes.md [hunk 4]
```diff
@@ -45,335 +45,48 @@ the case of `/`, the name `index`).
...
-    <td><code>/favs</code></td>
-    <td><code>favorites</code></td>
-    <td>↳<code>favorites.js</code></td>
-    <td>↳<code>favorites.js</code></td>
-    <td>↳<code>favorites.hbs</code></td>
-  </tr>
-</table>
-
-### Nested Routes
+In these cases, you can use **nested routes** to display one template inside
```
Comment (mixonic) [inline source/routing/defining-your-routes.md]: `you can nest routes to display`

Unsure nested routes is first-class enough to get bold etc.

File: source/routing/defining-your-routes.md [hunk 5]
```diff
@@ -45,335 +45,48 @@ the case of `/`, the name `index`).
...
-  <tr>
-    <td><code>/posts/new</code></td>
-    <td><code>posts.new</code></td>
-    <td>↳<code>posts.js</code><br>↳<code>posts/new.js</code></td>
-    <td>↳<code>posts.js</code><br>↳<code>posts/new.js</code></td>
-    <td>↳<code>posts.hbs</code><br>↳<code>posts/new.hbs</code></td>
-  </tr>
-</table>
+This router creates a route for `/posts` and for `/posts/new`. When a user
+visits `/posts`, they'll simply see the `posts.hbs` template. When the user
```
Comment (mixonic) [inline source/routing/defining-your-routes.md]: they do see the `posts/index` inside the `posts`.

Comment (mixonic) [inline source/routing/defining-your-routes.md]: I agree with the structural change to talk `index` in detail later, but I'd like it if this small intro can be accurate while not getting into the weeds.

File: source/routing/defining-your-routes.md (hunk 2 above)
Comment (michaelrkn) [inline source/routing/defining-your-routes.md]: I think it should be documented wherever we introduce the notion of the default behavior for a route like `this.route('post', { path: '/post/:post_id' });` loading the appropriate model. It has no context or meaning here.

**Comment by michaelrkn** [comment]
Addressed @mixonic's concerns, replied to @locks inline. Let me know if you two or anybody else has any other feedback!

**Comment by michaelrkn** [comment]
ping @locks @mixonic - if we're good here, i would love to get this merged in and keep going on my routing review :)

**Comment by locks** [comment]
:boom: :hammer:

**Pull request merged**